icefoganalytics/travel-authorization
# Lock Down Edits to Approved Travel Authorizations

# Context

I noticed that I could select and edit a Travel Authorization (TA) even after it was marked as approved. I'm unsure if this behavior is due to my system admin privileges or if there's an issue with how the approved data was initially created. We need to investigate further to determine if this is a genuine issue.

## Proposed Solution

- [x] (1) ~Implement enough basic test setup to test this endpoint.~ - too much work
- [x] (2) #66
- [x] (3) Lock down the UI so that non-admin users can see that they are not allowed to edit.
- [x] (4) Update the "Admin" UI such that admin user's know that they can only edit because they are admins. 
- [x] (5) Fix the "submit/save" button locations. The back button should be on every tab, but the "submit" and "save" buttons should only be on the "Details" tab.

### Task 3 Implementation

Add a read-only view for non-`draft` travel authorizations. Admins will see an "Edit" button that will redirect them to the appropriate edit page. e.g. `/admin/travel-authorization/37/edit`, this should be a completely different page than the "my-travel-request" route, specifically for admin edits. I will defer implementing this page for now.

### Task 4 Implementation

This will be handled by dint of using a different URL and view for edits performed as Admin user.

# Testing Instructions

1. Boot the app using `dev up`.
2. Log in to the app at http://localhost:8080/
3. Go to the My Travel Requests page via the top nav menu.
4. Create a new travel authorization.
5. Fill in all the fields, picking "Round Trip" trip type.
6. Submit the form.
7. Check that you are now redirected to the "read only" view of the submitted travel authorization.
8. Repeat this for each other "trip type".
9. Go to the QA Scenarios page via the top nav menu.
10. Apply the Admin or User role scenario and then go back to a read only form. If you are an Admin you get an extra (not fully implemented) edit button at the top of the page. If you are a User, you only get the default read only view.
